app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/WorkoutDetailsActivity.java

```diff
@@ -1,22 +1,147 @@
 package edu.drexel.lapcounter.lapcounter.frontend;
 
+import android.arch.lifecycle.ViewModelProviders;
+import android.content.Intent;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
+import android.widget.TextView;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.TimeZone;
 
 import edu.drexel.lapcounter.lapcounter.R;
+import edu.drexel.lapcounter.lapcounter.backend.Database.Device.Device;
+import edu.drexel.lapcounter.lapcounter.backend.Database.Device.DeviceViewModel;
+import edu.drexel.lapcounter.lapcounter.backend.TimestampConverter;
+import edu.drexel.lapcounter.lapcounter.backend.Database.Workout.WorkoutViewModel;
+import edu.drexel.lapcounter.lapcounter.backend.Database.Workout.Workout;
 import edu.drexel.lapcounter.lapcounter.frontend.navigationbar.NavBar;
 
 public class WorkoutDetailsActivity extends AppCompatActivity {
+
+
     private final NavBar mNavBar = new NavBar(this);
+    private WorkoutViewModel mWorkoutViewModel;
+    private DeviceViewModel mDeviceViewModel;
+    private int mWorkout_id;
+    private final static int MILLISECONDS_IN_SECOND = 1000;
+
+    private static String qualify(String s) {
+        return WorkoutDetailsActivity.class.getPackage().getName() + "." + s;
+    }
+    public static final String EXTRAS_WORKOUT_ID = qualify("DEVICE_ADDRESS");
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_workout_details);
 
+        Intent intent = getIntent();
+        mWorkout_id = intent.getIntExtra(EXTRAS_WORKOUT_ID,Integer.valueOf(getIntent().getStringExtra(EXTRAS_WORKOUT_ID)));
```

Comment:
Could you clarify why EXTRAS_WORKOUT_ID could be both a String and and int? it should only need to be one or the other.